### PR TITLE
User activation toggle

### PR DESCRIPTION
**Pull request by aphilipharrell**
Adds status filter and column for activating/deactivating users.

Relies on https://github.com/academic-innovation/mogc-partnerships/pull/33.

### CODE REVIEW

File: src/features/members/MemberList.jsx [hunk 1]
```diff
@@ -11,31 +13,56 @@ export default function MemberList({ cohort }) {
...
-            ],
+            filterChoices: statusFilterChoices,
+          },
+          {
+            Header: 'Action',
+            /* eslint-disable */
+            Cell: ({ row }) => (
+              <MemberActionRow member={row.values} cohort={cohort} />
+            ),
+            /* eslint-enable */
```
Comment (aphilipharrell) [inline src/features/members/MemberList.jsx:65]: ESLint flags this for recreating the component(s) on every render, but I don't know a way around that using the Paragon DataTable.

Comment (aphilipharrell) [reply src/features/members/MemberList.jsx:65]: 🤔 I wonder if building a list of components along with the member data would do it?

File: src/features/members/StatusChangeConfirmationModal.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ActionRow, Button, ModalDialog } from '@openedx/paragon';
+import { COHORT_MEMBERSHIP_STATUS, COHORT_MEMBERSHIP_STATUS_TYPE_MAP } from '../../utils/constants';
+
+export default function StatusChangeConfirmationModal({
+  isOpen, onClose, onSave, user,
+}) {
+  let statusText = 'deactivate';
+  let actionDescription = 'Deactivating a user will unenroll them in any courses they are currently enrolled in for this cohort.';
```
Comment (aphilipharrell) [inline src/features/members/StatusChangeConfirmationModal.jsx:10]: Need to verify the copy to use here.

File: src/features/members/MemberList.jsx [hunk 3]
```diff
@@ -11,31 +13,60 @@ export default function MemberList({ cohort }) {
...
-              { name: 'Invited', value: 'Invited' },
-            ],
+            filterChoices: statusFilterChoices,
+          },
+        ]}
+        additionalColumns={[
+          {
+            id: 'action',
+            Header: 'Action',
+            Cell: ({ row }) => (memberActionRows[row.values.id]),
```
Comment (aphilipharrell) [inline src/features/members/MemberList.jsx:69]: Edge-case: updating status of a member while a filter is applied will reset the data in the table so it is no longer filtered, but the filter selections will still be checked. To reset the filters, you have to click the checked filter, then click Clear Filters.

I haven't figured out a way around this, and am not sure if it is a bug in Paragon or how data is handled in this component. I tried memoizing `memberData` to no change.

**Comment by aphilipharrell** [comment]
[Partnership extension: Deactivation state](https://trello.com/c/wVUft6l6/848-partnership-extension-deactivation-state)

**Review by michaelwheeler — APPROVED** [review]

File: src/features/members/MemberList.jsx (hunk 3 above)
Comment (aphilipharrell) [reply src/features/members/MemberList.jsx:69]: I'm thinking it is probably a Paragon issue due to the similar behavior [noted in this QA card](https://trello.com/c/WC9JJLov/576-molp-admin-dashboard-clicking-clear-filters-doesnt-reset-the-selected-cohort-from-the-dropdown-menu-so-it-will-look-like-a-filte) but I will dig a bit more and see if I can find the source.

File: src/features/members/StatusChangeConfirmationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      title="Deactivate member"
+      isOpen={isOpen}
+      onClose={onClose}
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>Change member status</ModalDialog.Title>
+      </ModalDialog.Header>
+      <ModalDialog.Body>
+        <p>Are you sure you want to {statusText} {user.email}?</p>
+        <p>{actionDescription}</p>
```
[the remark this answers is not in the document]
Comment (aphilipharrell) [reply src/features/members/StatusChangeConfirmationModal.jsx:30]: Good call, I'll wrap in a condition.

**Review by michaelwheeler — APPROVED** [review]
Thanks @aphilipharrell! Merge at your convenience.

**Pull request merged**